main.go

```diff
@@ -38,45 +39,79 @@ var (
 	signingNameOverride = kingpin.Flag("name", "AWS Service to sign for").String();
 	hostOverride = kingpin.Flag("host", "Host to proxy to").String();
 	regionOverride = kingpin.Flag("region", "AWS region to sign for").String();
+	configSets = kingpin.Flag("config-set", "Host-based configuration overrides for role-arn/name/host/region (encoded as yaml)").Short('c').Strings()
 )
 
-func main() {
-	kingpin.Parse()
-
-	log.SetLevel(log.InfoLevel)
-	if *debug {
-		log.SetLevel(log.DebugLevel)
-	}
 
+func getSigner(roleArn string) *v4.Signer {
```

Comment:
NIT: no need to use value here, you're copying the string when calling the function, just pass a string pointer